test/support/utils.h

```diff
@@ -930,6 +932,41 @@ get_new_kernel_params(KernelParams)
 }
 #endif //TEST_DPCPP_BACKEND_PRESENT
 
+template <typename T>
+typename ::std::enable_if_t<std::is_arithmetic_v<T>, void>
+generate_arithmetic_data(T* input, std::size_t size, std::uint32_t seed)
+{
+    std::default_random_engine gen{seed};
+    std::size_t unique_threshold = 75 * size / 100;
+    if constexpr (sizeof(T) < sizeof(short)) // no uniform_int_distribution for chars
+    {
+        std::uniform_int_distribution<int> dist(std::numeric_limits<T>::lowest(), std::numeric_limits<T>::max());
+        std::generate(input, input + unique_threshold, [&] { return T(dist(gen)); });
+    }
+    else if constexpr (std::is_integral_v<T>)
+    {
+        std::uniform_int_distribution<T> dist(std::numeric_limits<T>::lowest(), std::numeric_limits<T>::max());
+        std::generate(input, input + unique_threshold, [&] { return dist(gen); });
+    }
+    else
```

Comment:
I would suggest joining these two cases (941-950) into one, like this:

```
template <typename T>
using GT = std::conditional_t<sizeof(T) < sizeof(short), int, T>;
std::uniform_int_distribution<GT> ....
```

Reply:
Thanks, done.